fix a bug where the challenge is reset after asw_mission_restart

mithrand0 opened the pull request.

mithrand0 committed:
fix a bug where the challenge is reset after asw_mission_restart

geotavros commented:
rd_challenge ConVar has a callback UpdateMatchmakingTagsCallback(), which runs a lot of code inside of it. It needs to be reviewed to understand if it's safe to change rd_challenge like this and in what place. Sorry, but I don't have time right now to review it.

mithrand0 commented:
> rd_challenge ConVar has a callback UpdateMatchmakingTagsCallback(), which runs a lot of code inside of it. It needs to be reviewed to understand if it's safe to change rd_challenge like this and in what place. Sorry, but I don't have time right now to review it.

Thanks for pointing out UpdateMatchmakingTagsCallback(), I'll dive into it.

mithrand0 committed:
Revert "fix a bug where the challenge is reset after asw_mission_restart"

This reverts commit bd0ea0a9cd1390430dc69c65e8770217c7dcf533.

mithrand0 committed:
Revert "Merge pull request #926 from ReactiveDrop/bugfix/rd-challenge"

This reverts commit e60ea785a62705561bd66f08a1516574a6c20fc7, reversing
changes made to 3b7795af4f2cd034d19951334077ceac61accfb8.

mithrand0 committed:
adds a variable that allows monitoring challenge changes and transitions

mithrand0 committed:
add more states

mithrand0 commented:
I reverted pull request #926, and added a convar, allowing me to check for this specific state.
In my opinion, this should eventually be fixed correctly, but this way it allows me to continue my other work.

TLDR: the functionality is reverted to what is was before #926
